homeassistant/components/withings/__init__.py

```diff
@@ -132,6 +123,25 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
+@dataclass(slots=True)
+class WithingsData:
+    """Dataclass to hold withings domain data."""
+
+    measurement_coordinator: WithingsMeasurementDataUpdateCoordinator
+    sleep_coordinator: WithingsSleepDataUpdateCoordinator
+    bed_presence_coordinator: WithingsBedPresenceDataUpdateCoordinator
+    goals_coordinator: WithingsGoalsDataUpdateCoordinator
+
+    def coordinators(self) -> set[WithingsDataUpdateCoordinator]:
```

Comment:
I added this since I will also add the WithingsClient in a later PR since I need the raw client for returning calendar events

Reply:
It looks like these are always iterated linearly so a list would be slightly cheaper

Reply:
I will do some more direct lookups so I think I prefer this one

Reply:
You could also avoid building the set every time the function is called by saving it in the dataclass itself as a protected var in a https://docs.python.org/3/library/dataclasses.html#dataclasses.__post_init__

Reply:
Gave it a shot